fetching/detector.py

```diff
@@ -0,0 +1,67 @@
+from abc import ABC, abstractmethod
+import pathlib
+import logging
+
+import toml
+
+from .projio import Project
+from .strat import ApplicationStrategy, PyTestStrategy
+
+
+class TestDetector(ABC):
+    """
+    When given the path to a repo, attempts to detect a specific
+    testing suite and create the appropriate application strategy
+    if there is a match.
+    """
+
+    def __init__(self, project: Project):
```

Comment:
None-check is missing (or does mypy already handle this?)

Reply:
mypy handles this for us ^^

Reply:
I just checked and sadly, it doesn't :/
Please add a check.

Reply:
```bash
λ poetry run mypy .
fetching/projio.py:21: error: Argument 1 to "Project" has incompatible type "None"; expected "Path"
```

Could it be you tried to do this from an `__init__.py` or from a file in the testing folders?

Reply:
Oh, I see. Yeah, I tried it in a test file.